### PR TITLE
Fix 'Bad state: No element' - Empty Path Metrics

### PR DESCRIPTION
**Pull request by HSCOGT**
I ran into the problem of 'computeMetrics' returning an empty list, getting the error 'Bad state: No element' because the code uses 'first' and 'last' getters, therefore I was getting a grey screen in a production build.

Quick fix, to validate if the list is not empty, it probably requires some analysis as to why  'computeMetrics' is not returning any values.

### CODE REVIEW

File: lib/widget_arrows.dart [hunk 1]
```diff
@@ -148,75 +148,83 @@ class _ArrowPainter extends CustomPainter {
...
-        final paint = Paint()
-          ..color = widget.color
-          ..style = PaintingStyle.stroke
-          ..strokeCap = StrokeCap.round
-          ..strokeJoin = StrokeJoin.round
-          ..strokeWidth = widget.width;
-
-        canvas.drawPath(path, paint);
+
+        if (path != null) {
```
Comment (Schwusch) [inline lib/widget_arrows.dart:152]: nit: a reversed statement avoids indentation, e.g.:
```dart
if (path == null) return;

File: lib/widget_arrows.dart [hunk 2]
```diff
@@ -148,75 +148,83 @@ class _ArrowPainter extends CustomPainter {
...
-
-    final angleStart = pi - tipAngleStart;
-    final originalPosition = tan.position;
-
-    if (lastPathMetric.length > 10) {
-      final tanBefore =
-          lastPathMetric.getTangentForOffset(lastPathMetric.length - 5)!;
-      adjustmentAngle = _getAngleBetweenVectors(tan.vector, tanBefore.vector);
-    }
+    if (metrics.isNotEmpty) {
```
Comment (Schwusch) [inline lib/widget_arrows.dart:173]: Same here, a reversed statement makes the changes easier to read:
```dart
if (metrics.isEmpty) return null;

**Comment by Schwusch** [comment]
Thanks for the PR!
It would be nice to know why the metrics are empty, but it's not good with a grey screen either :) 
Just a few nits, looks good otherwise

**Comment by HSCOGT** [comment]
Hi @Schwusch, thanks for taking a look, comments have been addressed and pushed.

**Comment by Schwusch** [comment]
Perfect! I'll push a release as soon as possible

**Comment by Schwusch** [comment]
Pushed to Pub: https://pub.dev/packages/widget_arrows/versions/0.6.0

**Pull request merged**